sci-electronics/kicad/kicad-4.0.2-r4.ebuild

```diff
@@ -0,0 +1,211 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+PYTHON_COMPAT=( python2_7 )
+
+WX_GTK_VER="3.0"
+
+inherit cmake-utils flag-o-matic gnome2-utils python-single-r1 vcs-snapshot wxwidgets versionator xdg
+
+DESCRIPTION="Electronic Schematic and PCB design tools."
+HOMEPAGE="http://www.kicad-pcb.org"
+LIBCONTEXT_COMMIT="3d92a1a50f4749b5a92131a957c9615473be85b4"
+
+SERIES=$(get_version_component_range 1-2)
+
+SRC_URI="https://launchpad.net/${PN}/${SERIES}/${PV}/+download/${P}.tar.xz
+	!minimal? (
+		http://downloads.kicad-pcb.org/libraries/${PN}-footprints-${PV}.tar.gz
+		https://github.com/KiCad/${PN}-library/archive/${PV}.tar.gz -> ${P}-library.tar.gz
+	)
+	i18n? ( https://github.com/KiCad/${PN}-i18n/archive/${PV}.tar.gz -> ${P}-i18n.tar.gz )
+	https://github.com/twlostow/libcontext/archive/${LIBCONTEXT_COMMIT}.tar.gz -> ${PN}-libcontext.tar.gz"
+
+LICENSE="GPL-2+ GPL-3+ Boost-1.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="debug doc examples github i18n libressl minimal +python webkit"
+LANGS="bg ca cs de el es fi fr hu it ja ko nl pl pt ru sk sl sv zh_CN"
+for lang in ${LANGS} ; do
+	IUSE="${IUSE} linguas_${lang}"
+done
+unset lang
+unset LANGS
+
+REQUIRED_USE="
+	github? ( webkit )
+	python? ( ${PYTHON_REQUIRED_USE} )"
+
+CDEPEND="x11-libs/wxGTK:${WX_GTK_VER}[X,opengl,webkit?]
+	python? (
+		dev-python/wxpython:${WX_GTK_VER}[opengl,${PYTHON_USEDEP}]
+		${PYTHON_DEPS}
+	)
+	>=dev-libs/boost-1.56[nls,threads,python?]
+	github? (
+		libressl? ( dev-libs/libressl )
+		!libressl? ( dev-libs/openssl:0 )
+	)
+	media-libs/glew
+	media-libs/freeglut
+	media-libs/mesa
+	sys-libs/zlib
+	x11-libs/cairo"
+DEPEND="${CDEPEND}
+	doc? ( app-doc/doxygen )
+	i18n? ( virtual/libintl )
+	python? ( dev-lang/swig:0 )
+	app-text/dos2unix"
+RDEPEND="${CDEPEND}
+	sci-electronics/electronics-menu"
+
+pkg_setup() {
+	python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	xdg_src_prepare
+
+	# Add separated out libcontext files and patch source to use them
+	mkdir -p "${S}/common/system/" || die "mkdir failed"
+	mkdir -p "${S}/include/system/" || die "mkdir failed"
+	cp "${WORKDIR}/${PN}-libcontext/libcontext.cpp" "${S}/common/system/libcontext.cpp" || die "cp failed"
+	cp "${WORKDIR}/${PN}-libcontext/libcontext.h" "${S}/include/system/libcontext.h" || die "cp failed"
+	# Path source to use new "built in" libcontext. Also patch libcontext.cpp to have correct include file.
+	# Path must be applied after new libcontext files have been copied to the kicad source directory.
+	epatch "${FILESDIR}/${PN}-boost-context.patch"
+	# Patch python swig import fixer build script
+	epatch "${FILESDIR}/${PN}-swig-import-helper.patch"
+
+	# remove all the non unix file endings
+	find "${S}" -type f -name "*.desktop" | xargs -n1 dos2unix
```

Comment:
If you wish to do so, you can use edos2unix from eutils which would allow you to drop the dependency for dos2unix